packages/atcommons/include/atcommons/json.h

```diff
@@ -9,14 +9,16 @@ extern "C" {
 // - atcommons/json.h
 
 #if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__NetBSD__) || \
-    defined(_WIN32)
+    defined(_WIN32) || defined(CONFIG_IDF_TARGET_ESP32)
 #define ATCOMMONS_JSON_PROVIDER_CJSON
-#include <cJSON.h> // IWYU pragma: export
+#endif
 
-#elif defined(CONFIG_IDF_TARGET_ESP32)
-#define ATCOMMONS_JSON_PROVIDER_CJSON
+#if __has_include(<cJSON.h>)
+#include <cJSON.h> // IWYU pragma: export
+#elif __has_include(<cjson/cJSON.h>)
+#include <cjson/cJSON.h> // IWYU pragma: export
+#elif __has_include(<cjson.h>)
 #include <cjson.h> // IWYU pragma: export
```

Comment:
using if __has_include macro to conditionally pull in the cJSON include since the location changes depending on how cjson was built & included